lib/graphql/batch/loader.rb

```diff
@@ -1,21 +1,20 @@
 module GraphQL::Batch
   class Loader
-    def self.for(*group_args)
-      loader_key = loader_key_for(*group_args)
-      executor = Executor.current
-
-      unless executor
-        raise GraphQL::Batch::NoExecutorError, 'Cannot create loader without'\
-          ' an Executor. Wrap the call to `for` with `GraphQL::Batch.batch`'\
-          ' or use `GraphQL::Batch::Setup` as a query instrumenter if'\
-          ' using with `graphql-ruby`'
+    # Use new argument forwarding syntax if available as an optimization
+    if RUBY_ENGINE && Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")
+      class_eval(<<~RUBY, __FILE__, __LINE__ + 1)
+        def self.for(...)
+          current_executor.loader(loader_key_for(...)) { new(...) }
+        end
+      RUBY
+    else
+      def self.for(*group_args, **group_kwargs)
+        current_executor.loader(loader_key_for(*group_args)) { new(*group_args) }
```

Comment:
`group_kwargs` isn't passed on here?

Reply:
Oops, I meant to remove it from the parameters so that it got captured in `*group_args`.  Pushed a fix